core/internal/cltest/cltest.go

```diff
@@ -1558,6 +1596,7 @@ func (hb *HeadBuffer) Append(head *eth.Head) {
 		ParentHash: head.ParentHash,
 		Parent:     head.Parent,
 		Timestamp:  time.Unix(int64(len(hb.Heads)), 0),
+		EVMChainID: utils.New(head.EVMChainID),
```

Comment:
Why New here and not using the existing pointer?

Reply:
Making the safe choice since this field was involved in the race. I'll revisit to see if we can be sure it's not mutated.

Reply:
AFAIK we never mutate a chain ID anywhere